Exiled.Events/Patches/Generic/HazardListAdd.cs

```diff
@@ -0,0 +1,44 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="HazardListAdd.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Generic
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using Exiled.API.Features.Hazards;
+    using Exiled.API.Features.Pools;
+    using HarmonyLib;
+    using Hazards;
+
+    /// <summary>
+    /// Patches <see cref="EnvironmentalHazard.Start()"/>.
+    /// </summary>
+    [HarmonyPatch(typeof(EnvironmentalHazard), nameof(EnvironmentalHazard.Start))]
+    internal class HazardListAdd
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+
+            newInstructions.InsertRange(
+                newInstructions.Count - 1,
```

Comment:
Would be able to use Postfix but that not a problem we can keep it transpiler

Reply:
postfix for the win